src/Parser.cpp

```diff
@@ -83,16 +85,72 @@ auto Parser::ParseRecordDef() -> p<DefinitionAST> {
   auto record_tok = expect(TokRecord);
   if (!record_tok)
     return {nullptr, NONCOMMITTED};
-  auto id = expect(TokID);
 
+  auto id = expect(TokID);
   if (!id) {
-
-    this->add_error(record_tok->location,
+    this->add_error(id->location,
                     "Failed to parse an identifier after token Record");
     return {nullptr, COMMITTED_NO_MORE_ERROR};
   }
-  // Julius continue from here :)
-  sammine_util::abort();
+
+  auto left_curly = expect(TokLeftCurly);
+  if (!left_curly) {
+    this->add_error(left_curly->location,
+                    fmt::format("Failed to parse the left curly braces for "
+                                "record definition after identifier {}",
+                                id->lexeme));
+    return {nullptr, COMMITTED_NO_MORE_ERROR};
+  }
+
+  std::vector<std::unique_ptr<TypedVarAST>> record_members;
+  while (!tokStream->isEnd()) {
+    auto [member, result] = ParseTypedVar();
+    switch (result) {
+    case SUCCESS: {
+      record_members.push_back(std::move(member));
+      // TODO:
+      // Later in the future we have to find a way to compose Record
+      // where the last member not needing a comma(,)
+      //
+      // name_1 is last in this case `Record id { name_1 };`
+      // name_2 is last in this case `Record id { name_1, name_2 };`
+      //
+      // for now we'll stick to `Record id { name_1, name_2, };`
+      if (!expect(TokComma)) {
+        this->add_error(
+            member->get_location(),
+            fmt::format("Failed to parse a Semi colon after the Identifier {}",
+                        member->name));
+        return {std::make_unique<RecordDefAST>(id, std::move(record_members)),
+                COMMITTED_NO_MORE_ERROR};
+      }
+    }
+
+    case NONCOMMITTED:
+    case COMMITTED_NO_MORE_ERROR:
+
+    case COMMITTED_EMIT_MORE_ERROR: {
+      this->add_error(
+          record_tok->location,
+          fmt::format("Failed to parse record {}", record_tok->lexeme));
+      return {std::make_unique<RecordDefAST>(id, std::move(record_members)),
+              COMMITTED_NO_MORE_ERROR};
+    } break;
+    }
+  }
+
+  auto right_curly = expect(TokRightCurly);
+  if (!right_curly) {
+    this->add_error(right_curly->location,
```

Comment:
segfaulting on right curly

Reply:
silly me, what would i change it instead in this case?
```suggestion
    auto last_member = record_members.back();
    this->add_error(last_member->get_location(),
```

i think we'd need a bit more code than that, in order to check to see if we've encountered a `Record id {};` or even with a regular definition `Record id { name_1, }`, we'd have to include the comma at after of the member `name_1`. So i'm assuming we'd have to do something like `last_member->get_location() + 1` to consume the comma , but just for the sake of getting the gist of it